src/test/java/org/cactoos/io/ZipTest.java

```diff
@@ -24,51 +24,121 @@
 
 package org.cactoos.io;
 
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.io.File;
+import java.io.OutputStream;
 import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
-import org.hamcrest.MatcherAssert;
-import org.hamcrest.Matchers;
+import java.util.zip.ZipFile;
+import org.cactoos.Text;
+import org.cactoos.iterable.IterableOf;
+import org.cactoos.iterable.Sticky;
+import org.cactoos.iterator.Mapped;
+import org.cactoos.list.ListOf;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasValues;
 
 /**
  * Test case for {@link Zip}.
  *
  * @since 0.29
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public final class ZipTest {
+
     /**
      * Temporary folder.
      */
     @Rule
-    public final TemporaryFolder folder = new TemporaryFolder();
+    public final TemporaryFolder temporal = new TemporaryFolder();
+
+    @Test
+    public void mustZipEmptyDirectory() throws Exception {
+        final File folder = this.temporal.newFolder("empty");
+        new Assertion<>(
+            "Must zip empty directory",
+            new Sticky<>(
+                new IterableOf<>(
```

Comment:
@fanifieiev why do you add even more decorators? I propose the following:
- returns a `SetOf` inside the scalar, so that we don't bother about order
- use the matcher `IsEqual` with a `SetOf` there too

And it should work perfectly.

We don't want to use `HasValues` because it is not strict about the content, it only check that some things are present (like `contains()`)

Reply:
@victornoel I need to wrap scalar lambda into iterable, that is why I am doing:
```
new IterableOf<>(
   () -> {...}
)
```

and I use `Sticky `because I need cached value.

Reply:
@fanifieiev I'm sorry, but that's way too complex. We should be able to simplify much more this code... what's the point of creating a list of a mapped iterator if after that you return the list's iterator and then you wrap it into an iterable that you make sticky? That's way too complex.

We spent already too much time here, please simplify that with as less as possible classes involved. Take the time to understand what's the expected behaviour of those classes and remove layers instead of adding more.

I see two simple solutions:
- the one you had before based on ScalarHasValue with a Sticky scalar responsible of holding the value and the lambda of the scalar just returning a `ListOf` or a `SetOf` (whichever, it's not important)
- move all this code in the lambda OUT of the assertion and simply store the resulting list or iterable or set or whatever in a variable.

Reply:
@fanifieiev I have created #1334 to explain what is the problem with `iterable.Sticky`.

Reply:
@victornoel OK